.github/workflows/ci.yml

```diff
@@ -55,7 +56,14 @@ jobs:
           ASAN_OPTIONS: "detect_leaks=0"
         run: |
           export LD_PRELOAD=`gcc -print-file-name=libasan.so`
-          py.test -s -m "not mypy and not pyright"
+          coverage run -m pytest -s -m "not mypy and not pyright"
+          coverage xml
+          gcov -abcu `find build/ -name *.o`
+
+      - name: Upload Codecov
+        uses: codecov/codecov-action@v3
+        with:
+          files: coverage.xml,_core.c.gcov,atof.h.gcov,ryu.h.gcov
```

Comment:
This uses the github actions codecov action to upload coverage files to codecov.io.

We manually specify the list of files, since `gcov` generates a few additional files based on inline functions found in the cpython header files (e.g. one for `cpython/Include/object.h`). There may be a cleaner way of doing this, but `msgspec` only has 3 C source files, so manually listing them seems fine.